pkg/common/table_info_helper.go

```diff
@@ -641,17 +653,16 @@ func (s *columnSchema) initIndexColumns() {
 		}
 	}
 
-	if handleIndexOffset < 0 {
+	if handleIndexOffset < 0 || len(s.HandleKeyIDs) > 0 {
```

Comment:
It's better add some comment here, it looks this code implies something not obvious.